erigon-lib/downloader/README.md

```diff
@@ -49,7 +49,31 @@ As this file ise versioned as part of the Erigon release process the file to has
 
 This is a file which resides in the <data-dir>/snapshots directory for an Erigon node.  It is created when the node performs its initial download.  It contains the list of downloaded files and their respective hashes.
 
-When a `snapshot-lock` file exists it is used reather than the chain.toml file to determine which files should be downloaded.  This means that the directory contents can be maintained event if Erigon is re-versioned and the chain.toml contents change.
+When a `snapshot-lock` file exists it is used instead than the chain.toml file to determine which files should be downloaded.  This means that the directory contents can be maintained event if Erigon is re-versioned and the chain.toml contents change.
+
+### Data Model
+
+#### downloads
+
+With E3 downloaded snapshots come in two forms, `fixed` where the file is immutable and should not be changed by the downloading node, and `baseline` where the downloaded file is used as a baseline which is then altered by the local node. These have the following operating requirements.
+
+##### `fixed`
+
+Fixed downloads are expected to be immutable which means that the file contents and thier associated hash are expected not to change.  On order to prevent disruption on the downloading node the hash of a fixed download is read from the pre-verfied hashes in `chains.toml` when a process is first started and written to the downloads section of `snapshot-lock`.   
+
+After this the process will attempt to download the hashed file from the torrent and once it is downloaded will re-check the file contents for consistency on each process restart.  If the file is inconsistent with the `snapshot-lock` hash it will be re-downloaded.
```

Comment:
1. `will re-check the file contents for consistency on each process restart` - which file? if data-file - why/how need check on each restart? 

2. `If the file is inconsistent with the `snapshot-lock` hash it will be re-downloaded` - in another words "snapshot-lock is new source of truth" - what is wrong with current source of truth? what problem you solving? 

3. Can solve points 1 and 2 by removing `ts.DisableInitialPieceCheck = true` line: https://github.com/ledgerwatch/erigon/blob/devel/erigon-lib/downloader/util.go#L300 ? 

5. `preserve read performance via the file's page-cache` - not only because of this. also because re-read 10Tb of data on every startup it's "too paranoid" mode. to ensure consistency we have next toolbox: 

4.1. ACID-db: has some tricks here https://github.com/ledgerwatch/erigon/blob/1879c764c1dd0be566f1b42371d1fbe14e3f2a00/erigon-lib/downloader/mdbx_piece_completion.go#L75 and https://github.com/ledgerwatch/erigon/blob/1879c764c1dd0be566f1b42371d1fbe14e3f2a00/erigon-lib/downloader/downloader.go#L2106 and https://github.com/ledgerwatch/erigon/blob/1879c764c1dd0be566f1b42371d1fbe14e3f2a00/erigon-lib/downloader/downloader.go#L1161

4.2 fsync on files. It's here https://github.com/anacrolix/torrent/blob/59ec9d6dd211210d30edc31e3bb1ca2b8785fb50/torrent.go#L2142 and here is the `Flush` method of `mmap` backend (we using it now): https://github.com/anacrolix/torrent/blob/59ec9d6dd211210d30edc31e3bb1ca2b8785fb50/storage/mmap.go#L75 aaand i don't see `Flush` method in `non-mmap backend`:  https://github.com/anacrolix/torrent/blob/59ec9d6dd211210d30edc31e3bb1ca2b8785fb50/storage/file.go#L109

4.3. Check hash once must be enough - but maybe we do "less then once" because of `ts.DisableInitialPieceCheck = true`